.changeset/twenty-walls-sit.md

```diff
@@ -0,0 +1,6 @@
+---
+"@pnpm/config": major
```

Comment:
@zkochan let me know if you'd like to make this a lower severity—I wasn't sure. Technically it's a behavioral change but I'm not sure it warrants a major version bump since it should only really affect people who are mid-migration.